codes/csharp/chapter_backtracking/preorder_find_constrained_paths.cs

```diff
@@ -0,0 +1,61 @@
+﻿/**
+ * File: preorder_find_constrained_paths.cs
+ * Created Time: 2023-04-17
+ * Author: hpstory (hpstory1024@163.com)
+ */
+
+using hello_algo.include;
+using NUnit.Framework;
+
+namespace hello_algo.chapter_backtracking;
+
+public class preorder_find_constrained_paths
+{
+    static List<TreeNode> path;
+    static List<List<TreeNode>> res;
+
+    /* 前序遍历 */
+    static void preOrder(TreeNode root)
+    {
+        // 剪枝
+        if (root == null || root.val == 3)
+        {
+            return;
+        }
+        // 尝试
+        path.Add(root);
+        if (root.val == 7)
+        {
+            // 记录解
+            res.Add(new List<TreeNode>(path));
+        }
+        preOrder(root.left);
+        preOrder(root.right);
+        // 回退
+        path.RemoveAt(path.Count - 1);
+    }
+
+    [Test]
+    public void Test()
+    {
+        TreeNode root = TreeNode.ArrToTree(new int?[] { 1, 7, 3, 4, 5, 6, 7 });
```

Comment:
Can I change this method(ArrToTree -> listToTree) to be the same signature as other languages? If possible, I will create a new PR to modify it.

Reply:
LGTM! Let's change it.

Reply:
Don't hesitate to introduce more ideas. Thanks!